ipam/ring/ring.go

```diff
@@ -348,6 +348,25 @@ func (r *Ring) OwnedRanges() (result []address.Range) {
 	return r.splitRangesOverZero(result)
 }
 
+// OwnedAndMergedRanges does the same as OwnedRanges, except that it returns
+// merged ranges.
+func (r *Ring) OwnedAndMergedRanges() (result []address.Range) {
```

Comment:
I'd move the merging to `HandleUpdate` not only because of the safety reason, but also because merging is needed only for awsvpc.